### PR TITLE
chore: make it so list_permitted and list_authorized are called as update calls

### PR DESCRIPTION
**Pull request by dfx-json**
# Description

This requires the `list_authorized` and `list_permitted` methods to be called as an update and disables the ability to call them as query calls. This resolves a potential security risk.

Fixes # (issue) 
FOLLOW-895

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -213,27 +256,37 @@ Existing identities are not affected by this change.
 
 ### fix: Certification for aliasing updates on asset deletion
 
-Best explained by an example: Two assets exist with aliasing enabled: `/content` and `/content.html`. Usually, when requesting `/content`, `/content.html` is served because it has aliasing enabled.
-But in this scenario, because `/content` exists, it overwrites the alias and `/content` is served when requesting the path `/content`.
+Best explained by an example: Two assets exist with aliasing enabled: `/content` and `/content.html`. Usually, when
```
Comment (ericswanson-dfinity) [inline CHANGELOG.md:259]: Is all this reformatting of the changelog intentional?

Comment (dfx-json) [reply CHANGELOG.md:259]: was not. my ide does this automatically when I commit. I'll revert

File: src/canisters/frontend/ic-certified-assets/src/lib.rs [hunk 2]
```diff
@@ -104,14 +104,14 @@ async fn validate_revoke_permission(arg: RevokePermissionArguments) -> Result<St
...
-#[candid_method(query)]
+#[update]
+#[candid_method(update)]
 fn list_authorized() -> ManualReply<Vec<Principal>> {
     STATE.with(|s| ManualReply::one(s.borrow().list_permitted(&Permission::Commit)))
 }
 
-#[query(manual_reply = true)]
-#[candid_method(query)]
+#[update]
```
Comment (sesi200) [inline src/canisters/frontend/ic-certified-assets/src/lib.rs:113]:
```suggestion
#[update(manual_reply = true)]
```
If you don't want to rewrite how the output is returned please keep `manual_reply`

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,24 +16,29 @@ The message "transaction is a duplicate of another transaction in block ...", pr
...
 # 0.15.1
 
 ### feat: Added support for reserved_cycles and reserved_cycles_limit
 
 `dfx canister status` will now display the reserved cycles balance and reserved cycles limit for a canister.
 
 Added command-line options:
-  - `dfx canister create --reserved-cycles-limit <limit>`
-  - `dfx canister update-settings --reserved-cycles-limit <limit>`
+- `dfx canister create --reserved-cycles-limit <limit>`
```
Comment (dfx-json) [inline CHANGELOG.md:40]: at this point, i'm not sure how to get my IDE to stop doing this formatting..

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**